app/controllers/multipage_contents_controller.rb

```diff
@@ -0,0 +1,31 @@
+class MultipageContentsController < ApplicationController
+  def show
+    content_item_response = ContentStore.service.content_item(base_path)
+    @content = model_class.new(content_item_response.to_hash) if content_item_response
+
+    if @content.present?
+      if params[:part]
+        @content.current_part = current_part
+        redirect_to base_path unless @content.current_part
+      end
+    else
+      render_404
+    end
+  end
+
+private
+
+  def model_class
+    controller_name.classify.constantize
+  end
+
+  def base_path
+    "/#{params[:slug]}"
+  end
+
+  def current_part
```

Comment:
Good point, this started as a controller instance var prior to the addition of `@content.current_part`, will update.